### ISSUE TITLE
lighttable remains in main view when enter darkroom via preview

### ISSUE THREAD

**Issue by dtorop** [darktable-org/darktable#5398]
**Describe the bug**

It's possible to enter darkroom view such that the main view shows lighttable view.

**To Reproduce**

   1. select lighttable view
   2. choose a collection with a few images
   3. select "file manager" or "zoomable light table" mode
   4. click to select an image
   5. press and hold "w" to preview the image
   6. while holding the "w" either press "d" or double-click the image to enter darkroom view
   7. though in darkroom view, the main view will show either the file manager or zoomable light table.

Initially darkroom mode will show the lighttable main view full screen. Releasing "w" shows the side panels. It's possible to return to lighttable view (type "L" or choose "lighttable" in top right"), and from there re-enter darkroom (not via preview) to work with an image as usual.

Note that if, while in darkroom, I double-click another image in the main area, darktable will hang. Last messages to console (with -d perf on) are:

```
788.632766 [dev_process_image] pixel pipeline processing took 0.121 secs (0.377 CPU)
789.387583 [dev] took 0.000 secs (0.000 CPU) to load the image.
789.392352 [dev_process_image] pixel pipeline processing took 0.005 secs (0.004 CPU)
[image_cache_allocate] failed to open image 4294967295 from database: no more rows available
815.346777 [dev] took 0.002 secs (0.002 CPU) to load the image.
```

Somewhat wonderfully, it's still possible to use the modules in the right panel to edit the image, which will update its thumbnail in the main area.

**Expected behavior**

When entering darkroom view from preview, the selected image should be the only one in the main area (no lighttable widgets).

**Screenshots**

![image](https://user-images.githubusercontent.com/2311860/84339436-74a4da80-ab6c-11ea-8c05-fe980d2709d3.png)

**Platform (please complete the following information):**
 - Darktable Version: master f6842cc0d100eb44079394566ab07558259fd9c6
 - OS: Debian bullseye/sid
 - OpenCL: activated
 - NVIDIA Corporation GP104GLM [Quadro P4000 Mobile] (rev a1), 4.6.0 NVIDIA 440.82

**Additional context**

- I've noticed this issue for a few months -- since the first rewrite of lighttable widgets. I only now got around to reproducing it.
- Pressing "w" in darkroom mode makes the image full screen, even when things are functioning as usual. Is this intentional, or is a binding that isn't released?

**Comment by AlicVB** [comment]
Nice effect, isn't it ;)
it's indeed certainly due to my lighttable rewrite. I'll have a look and fix asap.



>     * Pressing "w" in darkroom mode makes the image full screen, even when things are functioning as usual. Is this intentional, or is a binding that isn't released?

That's intended, there's also a full preview mode in darkroom for quite some time now (I think it was present in 2.6 already)
This is quite handy to have a quick look at your work without the distractions of side panels.

**Comment by dtorop** [comment]
> Nice effect, isn't it ;)

Indeed! This is our chance to move toward a truly modeless darktable!

> That's intended, there's also a full preview mode in darkroom for quite some time now (I think it was present in 2.6 already)
> This is quite handy to have a quick look at your work without the distractions of side panels.

I had no idea... Now I notice that it's there as a darkroom shortcut.

**Comment by TurboGit** [comment]
A nice one indeed :)

**Comment by dtorop** [comment]
@AlicVB I'm still seeing the lighttable-in-darkroom bug after 008c163c166e2a77fe809669a6f83cd5dbed1e72, unless I've gotten mixed up in pulling in the change. Here it's easy to reproduce:

1) select image in lighttable
2) press and hold "w"
3) press and release "d"

The result is something like:

![image](https://user-images.githubusercontent.com/2311860/84420597-4664e080-abe8-11ea-80c6-5adf30991aad.png)

008c163c166e2a77fe809669a6f83cd5dbed1e72 does fix entering darkroom with preview mode on!

**Comment by AlicVB** [comment]
At least I've fixed one part of the problem...
For the second part, I know what happen, (I've not reproduced because I show the filmstrip in my dev install)
I'll come with a fix asap

**Comment by AlicVB** [comment]
should be fixed by PR #5427 ...
